in_toto/in_toto_run.py

```diff
@@ -40,106 +74,47 @@
 log = logging.getLogger("in_toto")
 
 
-def in_toto_run(step_name, material_list, product_list, link_cmd_args,
-     record_streams, signing_key, gpg_keyid, gpg_use_default, gpg_home):
-  """
-  <Purpose>
-    Calls runlib.in_toto_run and catches exceptions
-
-  <Arguments>
-    step_name:
-            A unique name to relate link metadata with a step defined in the
-            layout.
-    material_list:
-            List of file or directory paths that should be recorded as
-            materials.
-    product_list:
-            List of file or directory paths that should be recorded as
-            products.
-    link_cmd_args:
-            A list where the first element is a command and the remaining
-            elements are arguments passed to that command.
-    record_streams:
-            A bool that specifies whether to redirect standard output and
-            and standard error to a temporary file which is returned to the
-            caller (True) or not (False).
-    signing_key:
-            If not None, link metadata is signed with this key.
-            Format is securesystemslib.formats.KEY_SCHEMA
-    gpg_keyid:
-            If not None, link metadata is signed with a gpg key identified
-            by the passed keyid.
-    gpg_use_default:
-            If True, link metadata is signed with default gpg key.
-    gpg_home:
-            Path to GPG keyring (if not set the default keyring is used).
-
-  <Exceptions>
-    SystemExit if any exception occurs
-
-  <Side Effects>
-    Calls sys.exit(1) if an exception is raised
-
-  <Returns>
-    None.
-  """
-
-  try:
-    runlib.in_toto_run(step_name, material_list, product_list, link_cmd_args,
-         record_streams, signing_key, gpg_keyid, gpg_use_default, gpg_home)
-
-  except Exception as e:
-    log.error("(in-toto-run) - {0}: {1}".format(type(e).__name__, e))
-    sys.exit(1)
-
 def main():
   """Parse arguments, load key from disk (prompts for password if key is
   encrypted) and call in_toto_run. """
 
   parser = argparse.ArgumentParser(
       description="Executes link command and records metadata")
-  # Whitespace padding to align with program name
-  lpad = (len(parser.prog) + 1) * " "
-
-  parser.usage = ("\n"
-      "%(prog)s  --step-name <unique step name>\n{0}"
-               "{{--key <functionary signing key path>, "
-               " --gpg [<functionary gpg signing key id>]}} \n{0}"
-               "[--gpg-home <path to gpg keyring>]\n{0}"
-               "[--materials <filepath>[ <filepath> ...]]\n{0}"
-               "[--products <filepath>[ <filepath> ...]]\n{0}"
-               "[--record-streams]\n{0}"
-               "[--no-command]\n{0}"
-               "[--verbose | --quiet] -- <cmd> [args]\n\n"
-               .format(lpad))
-
-  in_toto_args = parser.add_argument_group("in-toto options")
+
+  parser.usage = ("%(prog)s <named arguments> [optional arguments]"
```

Comment:
Is is better for the usage string to specify the required named arguments?

`(%s(prog)s --step-name foo --products ...)`

Reply:
That's what I had before, but it was messier to maintain. And the user can always use `--help` for a detailed description.